module-bsp/board/rt1051/bsp/eink/ED028TC1.cpp

```diff
@@ -782,33 +762,30 @@ EinkStatus_e EinkWaitTillPipelineBusy()
 
 EinkStatus_e EinkDitherDisplay()
 {
-    uint8_t cmdWithArgs[2] = {EinkDPC, EINK_DITHER_4BPP_MODE | EINK_DITHER_START};
+    std::uint8_t cmdWithArgs[2] = {EinkDPC, EINK_DITHER_4BPP_MODE | EINK_DITHER_START};
 
     if (BSP_EinkWriteData(cmdWithArgs, sizeof(cmdWithArgs), SPI_AUTOMATIC_CS) != 0) {
-        EinkResetAndInitialize();
         return EinkSPIErr;
     }
 
-    uint16_t flags = 0;
+    std::uint16_t flags = 0;
 
     s_EinkReadFlagsRegister(&flags);
 
-    // Wait for the dither operation finish
-    while (flags & EINK_FLAG_DITHER_IN_PROGRESS) {
-        vTaskDelay(pdMS_TO_TICKS(1));
-        s_EinkReadFlagsRegister(&flags);
```

Comment:
Is this while really unnecessary?

Reply:
It will stay forever in this loop so it is hard to notice if any error has occurred and the whole procedure and probably even a thread will be tucked.
Also, I didn't notice any situation where it succeeded in the second or the subsequent attempt.